pkg/apis/pipeline/v1beta1/pipeline_validation_test.go

```diff
@@ -2334,9 +2336,22 @@ func TestPipelineTasksExecutionStatus(t *testing.T) {
 			}},
 		}},
 		expectedError: apis.FieldError{
-			Message: `invalid value: pipeline tasks can not refer to execution status of any other pipeline task`,
+			Message: `invalid value: pipeline tasks can not refer to execution status of any other pipeline task or aggregate status of tasks`,
 			Paths:   []string{"tasks[0].params[bar-status].value"},
 		},
+	}, {
+		name: "invalid string variable in dag task accessing aggregate status of tasks",
+		tasks: []PipelineTask{{
+			Name:    "foo",
+			TaskRef: &TaskRef{Name: "foo-task"},
+			Params: []Param{{
+				Name: "tasks-status", Value: ArrayOrString{Type: ParamTypeString, StringVal: "$(tasks.status)"},
+			}},
+		}},
+		expectedError: apis.FieldError{
+			Message: `invalid value: pipeline tasks can not refer to execution status of any other pipeline task or aggregate status of tasks`,
+			Paths:   []string{"tasks[0].params[tasks-status].value"},
+		},
```

Comment:
please add a similar test with a dag task attempting to access the aggregate status of tasks in whenexpressions

Reply:
Have created a separate PR #3899 to validate when expressions in dag tasks such that the context variables are prohibited in the tasks.